src/location/index.js

```diff
@@ -33,18 +42,17 @@ class Location extends Component {
   }
 
   render() {
-    if (!this.state.tides && !this.state.weather) {
+    const { tides, weather, location } = this.state
+
+    if (!tides && !weather) {
```

Comment:
I think you want `||` right?

Reply:
This is to handle the first load, render the loading view until we have all data. Next loads will be from localStorage and updated async.

Reply:
right but it's `&&`, I think you want to show loading if either tides or weather are missing right?